SimPEG/DataMisfit.py

```diff
@@ -69,60 +33,108 @@ class l2_DataMisfit(BaseDataMisfit):
 
     .. math::
 
-        \mu_\\text{data} = {1\over 2}\left| \mathbf{W}_d (\mathbf{d}_\\text{pred} - \mathbf{d}_\\text{obs}) \\right|_2^2
+        \mu_\\text{data} = {1\over 2}\left|
+        \mathbf{W}_d (\mathbf{d}_\\text{pred} -
+        \mathbf{d}_\\text{obs}) \\right|_2^2
 
     """
 
+    std = 0.05  #: default standard deviation if not provided by survey
+    eps = None  #: default floor
+    eps_factor = 1e-5  #: factor to multiply by the norm of the data to create floor
+
     def __init__(self, survey, **kwargs):
         BaseDataMisfit.__init__(self, survey, **kwargs)
 
+        if self.std is None:
+            if getattr(self.survey, 'std', None) is not None:
+                print(
+                    'SimPEG.DataMisfit.l2_DataMisfit assigning default std '
+                    'of 5%'
+                )
+            else:
+                self.std = self.survey.std
+
+        if self.eps is None:
+            if getattr(self.survey, 'eps', None) is None:
+                print(
+                    'SimPEG.DataMisfit.l2_DataMisfit assigning default eps '
+                    'of 1e-5 * ||dobs||'
+                )
+                self.eps = (
+                    np.linalg.norm(Utils.mkvc(survey.dobs), 2)*self.eps_factor
+                )  # default
+            else:
+                self.eps = self.survey.eps
+
     @property
-    def Wd(self):
-        """getWd(survey)
+    def W(self):
+        """W
 
             The data weighting matrix.
 
             The default is based on the norm of the data plus a noise floor.
 
             :rtype: scipy.sparse.csr_matrix
-            :return: Wd
+            :return: W
 
         """
 
-        if getattr(self, '_Wd', None) is None:
+        if getattr(self, '_W', None) is None:
 
             survey = self.survey
+            self._W = Utils.sdiag(1/(abs(survey.dobs)*self.std+self.eps))
 
-            if getattr(survey,'std', None) is None:
-                print('SimPEG.DataMisfit.l2_DataMisfit assigning default std of 5%')
-                survey.std = 0.05
-
-            if getattr(survey, 'eps', None) is None:
-                print('SimPEG.DataMisfit.l2_DataMisfit assigning default eps of 1e-5 * ||dobs||')
-                survey.eps = np.linalg.norm(Utils.mkvc(survey.dobs),2)*1e-5
+        return self._W
 
-            self._Wd = Utils.sdiag(1/(abs(survey.dobs)*survey.std+survey.eps))
-        return self._Wd
-
-    @Wd.setter
-    def Wd(self, value):
-        self._Wd = value
+    @W.setter
+    def W(self, value):
```

Comment:
we should add a check that allows the user to input a vector and we cast it to a sparse diag matrix